### PR TITLE
Cyborgs drop their upgrades on the floor when reset or deconstructed

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Instead of being banished to the shadow realm for no reason, cyborg upgrades will now drop to the floor when a borg is reset or deconstructed.

## Why It's Good For The Game
There's really no reason why upgrades should be deleted when you do these things. It hurts the entire station because materials are wasted any time a borg either wishes/is ordered to change modules or has to be reconstructed, which hurts especially when mining is not delivering. If multiple borgs are involved, it can quickly become ruinously expensive to rebuild all their upgrades.

(Thanks @FunnyMan3595 for helping with this ::D)

## Testing
Spawned in, gave a borg every single upgrade, reset the borg. Upgrades dropped.
Put the upgrades back in. Deconstructed the borg. Upgrades dropped.

## Changelog
:cl:
tweak: Cyborgs now drop their upgrades when reset or deconstructed.
/:cl:

### CODE REVIEW

**Comment by Coolrune206** [comment]
Wouldn't people just start deconstructing and reconstructing cyborgs to reset their module then? As far as I know borgs are deliberately intended to lose upgrades when reset, but this would bypass that entirely.

**Comment by Fordoxia** [comment]
> Wouldn't people just start deconstructing and reconstructing cyborgs to reset their module then? As far as I know borgs are deliberately intended to lose upgrades when reset, but this would bypass that entirely.

Reset modules are unfathomably cheap. There is no indication in comments or anywhere else that the deletion was a deliberate design choice, and appears to have simply been coded that way just because.

**Review by 1080pCat — APPROVED** [review]
I dont personally see any issues

File: code/modules/mob/living/silicon/robot/robot_mob.dm [hunk 1]
```diff
@@ -514,6 +514,12 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 	SSblackbox.record_feedback("tally", "cyborg_modtype", 1, "[lowertext(selected_module)]")
 	notify_ai(2)
 
+/mob/living/silicon/robot/proc/spill_upgrades() // Take the borg's upgrades and spill them on the floor
```
Comment (Henri215) [inline code/modules/mob/living/silicon/robot/robot_mob.dm:517]:
```suggestion
/// Take the borg's upgrades and spill them on the floor
/mob/living/silicon/robot/proc/spill_upgrades()
```

**Review by Henri215 — APPROVED** [review]
Looks good to Mom

**Review by S34NW — APPROVED** [review]

**Pull request merged**